daemon/algod/server.go

```diff
@@ -358,6 +372,13 @@ func (s *Server) Start() {
 			fmt.Printf("netlistenfile error: %v\n", err)
 			os.Exit(1)
 		}
+
+		addrPort := getPortFromAddress(addr, s.log)
+		listenAddrPort := getPortFromAddress(listenAddr, s.log)
+
+		if addrPort == listenAddrPort {
+			s.log.Warnf("Warning: EndpointAddress port %v matches NetAddress port %v. This may lead to unexpected results when accessing endpoints.", addrPort, listenAddrPort)
```

Comment:
```suggestion
			s.log.Warnf("EndpointAddress port %v matches NetAddress port %v. This may lead to unexpected results when accessing endpoints.", addrPort, listenAddrPort)
```